controller/cmd/destination/main.go

```diff
@@ -75,11 +67,30 @@ func Main(args []string) {
 		}
 	}
 
+	enableEndpointSlices := global.GetEndpointSliceEnabled()
+
```

Comment:
I think we should call `EndpointSliceAccess` and error out eagerly if `enableEndpointsSlices` is set but we don't have access.

Reply:
I've resolved this, after we initialise the K8s.API if `enableEndpointSlices=true` but the resource is not registered we have a call to `log.Fatal`

Reply:
I think we probably want to move the EndpointSliceAccess check up to before initializing the k8s API.  If EndpointSlices are not present on the cluster then initializing the k8s API will hang at "Waiting for caches to sync" instead of getting to the fatal log line.